system-contracts/bootloader/bootloader.yul

```diff
@@ -23,31 +23,55 @@ object "Bootloader" {
 
             /// @dev This method ensures that the prices provided by the operator
             /// are not absurdly high
-            function validateOperatorProvidedPrices(l1GasPrice, fairL2GasPrice) {
-                if gt(l1GasPrice, MAX_ALLOWED_L1_GAS_PRICE()) {
-                    assertionError("L1 gas price too high")
+            function validateOperatorProvidedPrices(fairL2GasPrice, pubdataPrice) {
+                // The limit is the same for pubdata price and L1 gas price
+                if gt(pubdataPrice, MAX_ALLOWED_FAIR_PUBDATA_PRICE()) {
+                    assertionError("Fair pubdata price too high")
                 }
 
                 if gt(fairL2GasPrice, MAX_ALLOWED_FAIR_L2_GAS_PRICE()) {
                     assertionError("L2 fair gas price too high")
                 }
             }
 
-            /// @dev Returns the baseFee for this batch based on the
-            /// L1 gas price and the fair L2 gas price.
-            function getBaseFee(l1GasPrice, fairL2GasPrice) -> baseFee, gasPricePerPubdata {
-                // By default, we want to provide the fair L2 gas price.
-                // That it means that the operator controls
-                // what the value of the baseFee will be. In the future, 
-                // a better system, aided by EIP1559 should be added. 
+            /// @dev The overhead for a transaction slot in L2 gas. 
+            /// It is roughly equal to 80kk/MAX_TRANSACTIONS_PER_BATCH, i.e. how many gas would an L1->L2 transaction
+            /// need to pay to compensate for the batch being closed.
+            /// @dev It is expected of the operator to set the "fair L2 gas price" appropriately to ensure that it is 
+            /// compensated enough in case the batch might be prematurely sealed because of the transaction slots being filled up.
+            function TX_SLOT_OVERHEAD_GAS() -> ret {
+                ret := 10000
```

Comment:
Why it is not 8000 as described in the doc?

Reply:
In the doc it used to be 80k, but this PR also included the increase in the number of transactions in batch (1024 -> 10k) and so we overhead got reduced 10x (I decided to not go below 10k to take into account that the transaction's information will eventually be stored in the DB, etc, so it the overhead per transaction should not become much lower than Ethereum's 21k)